source/switch/ui.cpp

```diff
@@ -195,21 +201,27 @@ UIResult uiLoop(u32 keysDown) {
 				}
 			}
 		}
-
+		int dst;
 		for (int j = scroll; j < menuItemsCount; j++) {
-			u8 color = 255;
 			u32 h, w;
 			getTextDimensions(font16, menu[j], &w, &h);
 			u32 heightOffset = (40 - h) / 2;
 
 			if (i * separator + heightOffset + menuHSeparator > currentFBHeight - 85) continue;
```

Comment:
this might be the cause of the performance problem. Try using break instead of continue statement